src/rule.cpp

```diff
@@ -18,20 +18,25 @@ namespace ddwaf
 
 std::optional<event::match> condition::match_object(const ddwaf_object* object) const
 {
-    const bool hasTransformation        = !transformation.empty();
-    bool transformationWillChangeString = false;
+    // TODO: this might use mutexes internally, so it might be worth having a
+    //       flag to determine if the processor should be accessed atomatically
+    //       or not.
+    auto processor = std::atomic_load_explicit(&processor_, std::memory_order_relaxed);
```

Comment:
As discussed, I'll just use `std::memory_order_seq_cst` and add a flag to avoid using processors atomically when a rule / condition / processor is immutable.